src/main/java/org/spongepowered/common/SpongeImpl.java

```diff
@@ -140,4 +146,14 @@ public static Path getSpongeConfigDir() {
         return globalConfig;
     }
 
+    public static void postState(Class<? extends GameStateEvent> type, GameState state) {
+        getGame().setState(state);
+        ((SpongeEventManager) getGame().getEventManager()).post(SpongeEventFactoryUtils.createState(type, getGame()), true);
+    }
+    
+    public static void postShutdownEvents() {
+        postState(GameStoppingEvent.class, GameState.GAME_STOPPING);
```

Comment:
Mm good point we don't really have any actions during the destruction of the game.